tests/test_input.py

```diff
@@ -0,0 +1,673 @@
+                'use_col' : 'use',
+             }
+    input = piff.InputFiles(config, logger=logger)
+    assert len(input.image_pos) == 1
+    assert len(input.image_pos[0]) == 85
+
+    # Can do both
+    config = {
+                'dir' : 'input',
+                'image_file_name' : 'test_input_image_00.fits',
+                'cat_file_name' : 'test_input_cat_00.fits',
+                'use_col' : 'use',
+                'flag_col' : 'flag',
+             }
+    input = piff.InputFiles(config, logger=logger)
+    assert len(input.image_pos) == 1
+    assert len(input.image_pos[0]) == 68
+
+    # Check invalid column names
+    base_config = {
+                'dir' : 'input',
+                'image_file_name' : 'test_input_image_00.fits',
+                'cat_file_name' : 'test_input_cat_00.fits', }
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(x_col='xx', **base_config))
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(y_col='xx', **base_config))
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(sky_col='xx', **base_config))
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(gain_col='xx', **base_config))
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(flag_col='xx', **base_config))
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(use_col='xx', **base_config))
+
+    # Can't give duplicate sky, gain
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(sky_col='sky', sky=3, **base_config))
+    np.testing.assert_raises(ValueError, piff.InputFiles, dict(gain_col='gain', gain=3, **base_config))
+
+    # Invalid header keys
+    np.testing.assert_raises(KeyError, piff.InputFiles, dict(sky='sky', **base_config))
+    np.testing.assert_raises(KeyError, piff.InputFiles, dict(gain='gain', **base_config))
+
+
+@timer
+def test_chipnum():
+    """Test the ability to renumber the chipnums
+    """
+    if __name__ == '__main__':
+        logger = piff.config.setup_logger(verbose=2)
+    else:
+        logger = piff.config.setup_logger(log_file=os.path.join('output','test_input.log'))
+
+    # First, the default is to just use the index in the image list
+    dir = 'input'
+    image_files = 'test_input_image_*.fits'
+    cat_files = 'test_input_cat_*.fits'
+    config = {
+                'dir' : dir,
+                'image_file_name' : image_files,
+                'cat_file_name': cat_files
+             }
+    input = piff.InputFiles(config, logger=logger)
+    assert input.chipnums == list(range(3))
+
+    # Now make the chipnums something else
+    config = {
+                'dir' : dir,
+                'image_file_name' : image_files,
+                'cat_file_name': cat_files,
+                'chipnum' : [ 5, 6, 7 ]
+             }
+    input = piff.InputFiles(config, logger=logger)
+    assert input.chipnums == [ i+5 for i in range(3) ]
+
+    # Use the GalSim Eval capability to get the index + 1
```

Comment:
Is there an ability to get the chipnum either from a column in the header or in the file name? Or is the thought that if we wanted a DES PSF's chipnums to correspond to the ccd number, we would just input the list ourselves?